1.1 Desktop app calls Web API - PoP/TodoListClient/Program.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Microsoft.Identity.Client;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TodoListClient
+{
+    class Program
+    {
+        /// <summary>
+        /// Instance of Cloud
+        /// </summary>
+        private static readonly string AadInstance = ConfigurationManager.AppSettings["ida:AADInstance"];
+
+        /// <summary>
+        /// Tenant
+        /// </summary>
+        private static readonly string Tenant = ConfigurationManager.AppSettings["ida:Tenant"];
+
+        /// <summary>
+        /// ClientID of the application
+        /// </summary>
+        private static readonly string ClientId = ConfigurationManager.AppSettings["ida:ClientId"];
+
+        /// <summary>
+        /// Authority
+        /// </summary>
+        private static readonly string Authority = string.Format(CultureInfo.InvariantCulture, AadInstance, Tenant);
+
+        /// <summary>
+        /// Scope of the TodoList action
+        /// </summary>
+        private static readonly string TodoListScope = ConfigurationManager.AppSettings["todo:TodoListScope"];
+
+        /// <summary>
+        /// Base address of the todolist Web API
+        /// </summary>
+        private static readonly string TodoListBaseAddress = ConfigurationManager.AppSettings["todo:TodoListBaseAddress"];
+        private static readonly string[] Scopes = { TodoListScope };
+        private static string TodoListApiAddress
+        {
+            get
+            {
+                string baseAddress = TodoListBaseAddress;
+                return baseAddress.EndsWith("/") ? TodoListBaseAddress + "api/todolist"
+                                                 : TodoListBaseAddress + "/api/todolist";
+            }
+        }
+
+        static async Task Main(string[] args)
+        {
+            var app = PublicClientApplicationBuilder.Create(ClientId)
+                .WithAuthority(Authority)
+                .WithDefaultRedirectUri()
+                .WithExperimentalFeatures() // Needed for PoP
+                .Build();
+            TokenCacheHelper.EnableSerialization(app.UserTokenCache);
+
+            AuthenticationResult result;
+
+            HttpClient httpClient = new HttpClient();
+            HttpResponseMessage response;
+
+            while (true)
```

Comment:
This method is too long - can you split it into 2 private methods, maybe "WriteItem" and "ReadItem" ?

Reply:
Or better yet - create a little menu like here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/a9f1175dd747777bcacea5063c8211f94f1f0ec0/tests/devapps/NetFxConsoleTestApp/Program.cs#L159

Reply:
Let me know if you want me to do this for you.